api/service/stagedstreamsync/beacon_helper.go

```diff
@@ -55,7 +58,15 @@ func (bh *beaconHelper) start() {
 }
 
 func (bh *beaconHelper) close() {
-	close(bh.closeC)
+	bh.lock.Lock()
```

Comment:
Channels by default thread safe, no reason having additional lock here

Reply:
Can we have separate PR to remove it?

Reply:
minor, can be fixed in next patch